xesmf/frontend.py

```diff
@@ -895,6 +953,85 @@ def __init__(
         else:
             self.out_coords = {lat_out.name: lat_out, lon_out.name: lon_out}
 
+        if parallel:
+            # Ensure ds_in is not dask-backed
+            if xr.core.pycompat.is_dask_collection(ds_in):
+                ds_in = ds_in.compute()
+
+            # Drop unnecessary variables in ds_in to save memory
+            if not locstream_in:
+                # Drop unnecessary dims
+                ds_in_dims_drop = set(ds_in.cf.coordinates.keys()).difference(
+                    ['longitude', 'latitude']
+                )
+                ds_in = ds_in.cf.drop_dims(ds_in_dims_drop)
+
+                # Drop unnecessary vars
+                ds_in_coords = list(ds_in.coords.variables)
+                ds_in_variables = list(ds_in.variables)
+                ds_in_vars_drop = set(ds_in_variables).difference(ds_in_coords)
+                ds_in = ds_in.drop_vars(ds_in_vars_drop)
+
+            # if bounds in ds_out, we switch to cf bounds for map_blocks
+            if 'x_b' in ds_out.dims:
+                lon_name = ds_out.cf['longitude'].name
+                lat_name = ds_out.cf['latitude'].name
+                ds_out = ds_out.cf.add_bounds([lon_name, lat_name])
+                ds_out = ds_out.drop_dims(['x_b', 'y_b'])
```

Comment:
The name of the dimension varies a lot, there are no rules... Also, I don't think you want to re-compute the bounds, the cf-xarray method is only an approximation.

```suggestion
            if 'lon_b' in ds_out and (ds_out.lon_b.ndims == ds_out.cf['longitude'].ndims):
                dims = ds
                ds_out = ds_out.assign_coords(
                    lon_bounds=cfxr.vertices_to_bounds(ds_out.lon_b, ('bounds', *ds_out.cf['longitude'].dims)),
                    lat_bounds=cfxr.vertices_to_bounds(ds_out.lat_b, ('bounds', *ds_out.cf['latitude'].dims))
                )
                # Make cf-xarray aware of the new bounds
                ds_out.cf['longitude'].attrs['bounds'] = 'lon_bounds'
                ds_out.cf['latitude'].attrs['bounds'] = 'lat_bounds'
                ds_out = ds_out.drop_dims(ds_out.lon_b.dims + ds_out.lat_b.dims)
```